### ISSUE TITLE
Fails to start after new installation with rt-detr or yolo5 models...

### ISSUE THREAD

**Issue by jjw99** [xnorpx/blue-onyx#246]
I'm not off to a great start trying to use Blue Onyx.

After the default installation and the status stuck at "Initializing...", I ran it manually to check the status:

C:\Program Files\Blue Onyx>blue_onyx.exe --log-level debug
2025-11-09T20:42:11.538869Z  INFO blue_onyx: Logging initialized with dynamic filtering support log_level=Debug
2025-11-09T20:42:11.538948Z  INFO blue_onyx::system_info: System Information:
2025-11-09T20:42:11.539007Z  INFO blue_onyx::system_info: CPU | GenuineIntel | 12th Gen Intel(R) Core(TM) i5-12600K | 10 Cores | 16 Logical Cores
2025-11-09T20:42:11.553483Z  INFO blue_onyx::system_info: GPU: Intel(R) UHD Graphics 770
2025-11-09T20:42:11.557379Z  INFO blue_onyx::system_info: GPU: NVIDIA GeForce RTX 3050
2025-11-09T20:42:11.557481Z  INFO blue_onyx::cli: === Blue Onyx Configuration ===
2025-11-09T20:42:11.557521Z  INFO blue_onyx::cli: Server Configuration:
2025-11-09T20:42:11.557549Z  INFO blue_onyx::cli:   Port: 32169
2025-11-09T20:42:11.557578Z  INFO blue_onyx::cli:   Request timeout: 15 seconds
2025-11-09T20:42:11.557609Z  INFO blue_onyx::cli:   Worker queue size: auto-determined
2025-11-09T20:42:11.557639Z  INFO blue_onyx::cli: Model Configuration:
2025-11-09T20:42:11.557672Z  INFO blue_onyx::cli:   Detection model type: rf-detr
2025-11-09T20:42:11.557756Z  INFO blue_onyx::cli:   Model: default (rf-detr-n.onnx)
2025-11-09T20:42:11.557805Z  INFO blue_onyx::cli:   Object classes: default (coco_classes.yaml)
2025-11-09T20:42:11.557843Z  INFO blue_onyx::cli: Detection Configuration:
2025-11-09T20:42:11.557880Z  INFO blue_onyx::cli:   Confidence threshold: 0.50
2025-11-09T20:42:11.557917Z  INFO blue_onyx::cli:   Object filter: none (all objects)
2025-11-09T20:42:11.557947Z  INFO blue_onyx::cli: Performance Configuration:
2025-11-09T20:42:11.557975Z  INFO blue_onyx::cli:   Force CPU: no
2025-11-09T20:42:11.558006Z  INFO blue_onyx::cli:   GPU index: 0
2025-11-09T20:42:11.558036Z  INFO blue_onyx::cli:   Intra threads: 16
2025-11-09T20:42:11.558064Z  INFO blue_onyx::cli:   Inter threads: 16
2025-11-09T20:42:11.558101Z  INFO blue_onyx::cli: Logging Configuration:
2025-11-09T20:42:11.558128Z  INFO blue_onyx::cli:   Log level: Debug
2025-11-09T20:42:11.558160Z  INFO blue_onyx::cli:   Log path: stdout
2025-11-09T20:42:11.558192Z  INFO blue_onyx::cli: Output Configuration:
2025-11-09T20:42:11.558221Z  INFO blue_onyx::cli:   Save processed images: disabled
2025-11-09T20:42:11.558257Z  INFO blue_onyx::cli:   Save statistics: disabled
2025-11-09T20:42:11.558286Z  INFO blue_onyx::cli: === Configuration Complete ===
2025-11-09T20:42:11.558438Z  INFO blue_onyx: DirectML is available for GPU inference
2025-11-09T20:42:11.559064Z  INFO blue_onyx::system_info: GPU: Intel(R) UHD Graphics 770
2025-11-09T20:42:11.559099Z  INFO blue_onyx::system_info: GPU: NVIDIA GeForce RTX 3050
2025-11-09T20:42:11.559269Z  INFO blue_onyx::startup_coordinator: Startup worker thread: Beginning detector initialization...
2025-11-09T20:42:11.559304Z  INFO blue_onyx::server: Starting server, listening on 0.0.0.0:32169
2025-11-09T20:42:11.559345Z  INFO blue_onyx::server: Welcome page, http://127.0.0.1:32169
2025-11-09T20:42:11.559366Z  INFO blue_onyx::detector::onnx: DirectML available, attempting to use DirectML for inference gpu_index=0
2025-11-09T20:42:11.559408Z  INFO blue_onyx::detector::onnx: DirectML initialization successful
2025-11-09T20:42:11.559507Z  INFO blue_onyx: Model and YAML files ready: C:\Program Files\Blue Onyx\rf-detr-n.onnx and C:\Program Files\Blue Onyx\rf-detr-n.yaml
2025-11-09T20:42:11.595858Z  INFO blue_onyx::detector::onnx: Initializing detector with model: "rf-detr-n.onnx" and inference running on GPU
2025-11-09T20:42:12.271472Z  INFO blue_onyx::detector::onnx: Model 'rf-detr-n.onnx' configured with input size: 384x384 (384x384 tensor)
2025-11-09T20:42:12.275853Z  INFO blue_onyx::detector::onnx: Warming up the detector
2025-11-09T20:42:12.279241Z DEBUG blue_onyx::detector::onnx: Decode image time: 3.3108ms, resolution 768x576
2025-11-09T20:42:12.279306Z DEBUG blue_onyx::detector::onnx: Image resize factors: width_factor=2.000 (384->768), height_factor=1.500 (384->576)
2025-11-09T20:42:12.279372Z DEBUG blue_onyx::image: Resizing image from 768x576 to 384x384
2025-11-09T20:42:12.280267Z DEBUG blue_onyx::detector::onnx: Resize image time: 894.9µs
2025-11-09T20:42:12.280322Z DEBUG blue_onyx::detector::onnx: Resized image dimensions: 384x384, Input tensor dimensions: 384x384
2025-11-09T20:42:12.280360Z DEBUG blue_onyx::detector::onnx: Expected pixels: 147456, Actual pixels: 147456
2025-11-09T20:42:12.280382Z DEBUG blue_onyx::detector::onnx: Input tensor shape: [1, 3, 384, 384]
2025-11-09T20:42:12.281063Z DEBUG blue_onyx::detector::onnx: Copy pixels to input time: 702.8µs
2025-11-09T20:42:12.281437Z DEBUG blue_onyx::detector::onnx: Pre-process model input time: 327.6µs
2025-11-09T20:42:12.281736Z DEBUG blue_onyx::detector::onnx: Pre-process time: 5.8083ms
2025-11-09T20:42:12.309885Z DEBUG blue_onyx::detector::onnx: Inference time: 28.0169ms
2025-11-09T20:42:12.309994Z DEBUG blue_onyx::detector::onnx: Ignoring unexpected RF-DETR output unexpected_output="labels"
**2025-11-09T20:42:12.408896Z ERROR blue_onyx::startup_coordinator: Startup worker thread: Detector initialization failed error=RF-DETR output 'logits' missing**
2025-11-09T20:42:12.409019Z  INFO blue_onyx::startup_coordinator: Startup worker thread: Completed due to initialization failure
**2025-11-09T20:42:12.409034Z ERROR blue_onyx::server: Detector initialization failed error=RF-DETR output 'logits' missing**
2025-11-09T20:43:22.642902Z  INFO blue_onyx: Ctrl+C received, shutting down server
2025-11-09T20:43:22.643087Z  INFO blue_onyx: Global shutdown requested

Using the web server I tried to switch to the yolov5 model, and the api just returns a different error...
Error: Status: 500 Internal Server Error, Body: {"success":false,"message":"","error":"Detector initialization failed: YOLOv5 detections tensor must be 2D","predictions":[],"count":0,"command":"","moduleId":"","executionProvider":"","canUseGPU":false,"inferenceMs":0,"processMs":0,"analysisRoundTripMs":0}

**Comment by xnorpx** [comment]
Yes sorry, I was vibecoding to much and pushed a regression yesterday. I have reverted that change.

**Comment by xnorpx** [comment]
Please uninstall that version and take the one from before:

https://github.com/xnorpx/blue-onyx/releases/tag/v0.8.0-ca90b57

**Comment by jjw99** [comment]
Thanks, I just tried the previous version and everything is starting now.

**Comment by xnorpx** [comment]
Breaking change has been reverted, I will test my s*** better next time and not push Saturday evening :)
